### ISSUE TITLE
Passing `--threads 1` does not make test output sequential

### ISSUE THREAD

**Issue by donmccurdy** [vanillaes/tape-es#8]
Thanks for making this! I like tape's simplicity but have been frustrated with the _un_-simplicity of using the `tape` or `tap` packages with ES modules, this test runner fills a need there. 

I'd like to use `tape-es` with a formatter like [`tap-spec`](https://www.npmjs.com/package/tap-spec). Even with `--threads 1`, output is jumbled and so the reporting isn't useful. The example below has two test groups, but all output is grouped under the second:

```
$ tape-es test/**/*.test.js --threads 1 | tap-spec

  SceneBinding


  NodeBinding

    ✔ should be strictly equal
    ✔ scene → children → 3
    ✔ scene → 1 child
    ✔ node1 → name
    ✔ node1 → children
    ✔ scene → children → 2
    ✔ node1 → position
    ✔ node1 → quaternion
    ✔ node1 → scale
    ✔ node2 → name
    ✔ node2 → children
    ✔ node2 → position
    ✔ node2 → quaternion
    ✔ node2 → scale
    ✔ node1 → name
    ✔ node1 → position


  total:     16
  passing:   16
  duration:  264ms
```

I understand why this happens when running tests in parallel, but it shouldn't happen with `--threads 1`, I think?

**Comment by evanplaice** [comment]
Yep, single-or-multi threaded. There needs to be an orchestration layer that collects and collates results so they are printed in the correct order.

This runner works by creating a subprocess for each test. The tests get fired off and results get piped directly to stdout in the order that they complete. To make the results ordered, the result of stdout needs to be collected in the main thread and collated in the same order that the tests execute.

It'll look something like:
- create an object that maps path -> result
- result gets initialized to null/undefined
- run the tests and insert the results into the object as they complete
- after all tests complete, convert the object to an array
- sort the array by path
- print the array to stdout

I doubt this'll be too difficult to implement. I just haven't taken the time to sit down and work on it yet.

**Comment by evanplaice** [comment]
Fixed in v3.0.0
